test/unitTests/observers/errorMessageObserver.test.ts

```diff
@@ -0,0 +1,128 @@
+    let sinon: SinonSandbox;
+
+    let messageToShow: string;
+    let messageActions: MessageItem[];
+
+    let observer: ErrorMessageObserver;
+
+    before(() => {
+        observer = new ErrorMessageObserver();
+
+        sinon = createSandbox();
+
+        sinon
+            .stub(vscode.window, 'showErrorMessage')
+            .callsFake((message: string, ...items: any[]) => {
+                messageToShow = message;
+                messageActions = items;
+
+                return undefined;
+            });
+    });
+
+    beforeEach(() => {
+        messageToShow = undefined;
+        messageActions = [];
+    });
+
+    after(() => {
+        sinon.restore();
+    });
+
+    describe(`UserSignInCompleted`, () => {
+        beforeEach(() => {
+            messageToShow = undefined;
+            messageActions = [];
+        });
+
+        it(`Sign in failed`, () => {
+            let event = new UserSignInFailed(`fakedCorrelationId`, new Error("faked message"));
+            observer.eventHandler(event);
+            assert.equal(messageToShow, '[Docs Validation] Signing in failed: faked message');
+            assert.deepEqual(messageActions, []);
+        });
+
+        it(`Sign in succeeded`, () => {
+            let event = new UserSignInSucceeded(`fakedCorrelationId`, fakedCredential, true);
+            observer.eventHandler(event);
+            assert.equal(messageToShow, undefined);
+            assert.deepEqual(messageActions, []);
+        });
+    });
+
+    describe(`UserSignOutCompleted`, () => {
+        beforeEach(() => {
+            messageToShow = undefined;
+            messageActions = [];
+        });
+
+        it(`Sign out failed`, () => {
+            let event = new UserSignOutFailed(`fakedCorrelationId`, new Error("faked message"));
+            observer.eventHandler(event);
+            assert.equal(messageToShow, '[Docs Validation] Signing out failed: faked message');
+            assert.deepEqual(messageActions, []);
+        });
+
+        it(`Sign out succeeded`, () => {
+            let event = new UserSignOutSucceeded(`fakedCorrelationId`);
+            observer.eventHandler(event);
+            assert.equal(messageToShow, undefined);
+            assert.deepEqual(messageActions, []);
+        });
+    });
+
+    describe(`BuildCompleted`, () => {
+        beforeEach(() => {
+            messageToShow = undefined;
+            messageActions = [];
+        });
+
+        it(`Build succeeded`, () => {
+            let event = new BuildSucceeded(`fakedCorrelationId`, fakedBuildInput, 0, <BuildResult>{});
+            observer.eventHandler(event);
+            assert.equal(messageToShow, undefined);
+            assert.deepEqual(messageActions, []);
+        });
+
+        it(`Build failed since credential expires`, () => {
+            let event = new BuildFailed(`fakedCorrelationId`, fakedBuildInput, 0, new DocsError("faked message", ErrorCode.TriggerBuildWithCredentialExpired));

```

Comment:
minor suggestion: is it a better practice to use a global string variable fakeMessage = ‘faked message’? It's used several times in this test class.